File: src/C3.Catalogue/Tapes/TapeSide.vb
Namespace Tapes

    Public NotInheritable Class TapeSide

        Public Sub New(
                isRecorded As Boolean,
                name As String,
                recordedAt As DateTime,
                deckName As String,
                inputName As String,
                peakLevel As Integer,
                noiseReduction As String,
                hx As Boolean,
                mpx As Boolean,
                dubbed As Boolean,
                speed As String,
                bias As Integer,
                biasCalibration As Integer,
                equalization As String,
                level As Decimal,
                levelCalibration As Decimal,
                contents As String,
                artist As String,
                title As String)

            Me.IsRecorded = isRecorded
            Me.Name = name
            Me.RecordedAt = recordedAt
            Me.DeckName = deckName
            Me.InputName = inputName
            Me.PeakLevel = peakLevel
            Me.NoiseReduction = noiseReduction
            Me.Hx = hx
            Me.Mpx = mpx
            Me.Dubbed = dubbed
            Me.Speed = speed
            Me.Bias = bias
            Me.BiasCalibration = biasCalibration
            Me.Equalization = equalization
            Me.Level = level
            Me.LevelCalibration = levelCalibration
            Me.Contents = contents
            Me.Artist = artist
            Me.Title = title
        End Sub

        Public ReadOnly Property IsRecorded As Boolean
        Public ReadOnly Property Name As String
        Public ReadOnly Property RecordedAt As DateTime
        Public ReadOnly Property DeckName As String
        Public ReadOnly Property InputName As String
        Public ReadOnly Property PeakLevel As Integer
        Public ReadOnly Property NoiseReduction As String
        Public ReadOnly Property Hx As Boolean
        Public ReadOnly Property Mpx As Boolean
        Public ReadOnly Property Dubbed As Boolean
        Public ReadOnly Property Speed As String
        Public ReadOnly Property Bias As Integer
        Public ReadOnly Property BiasCalibration As Integer
        Public ReadOnly Property Equalization As String
        Public ReadOnly Property Level As Decimal
        Public ReadOnly Property LevelCalibration As Decimal
        Public ReadOnly Property Contents As String
        Public ReadOnly Property Artist As String
        Public ReadOnly Property Title As String

        Public Shared Function Empty() As TapeSide
            Return New TapeSide(
                False,
                String.Empty,
                DateTime.MinValue,
                String.Empty,
                String.Empty,
                0,
                String.Empty,
                False,
                False,
                False,
                String.Empty,
                0,
                0,
                String.Empty,
                Decimal.Zero,
                Decimal.Zero,
                String.Empty,
                String.Empty,
                String.Empty)
        End Function

    End Class

End Namespace

File: src/C3.Catalogue/Tapes/TapeSide.cs
using System;

namespace C3.Catalogue.Tapes
{
    public sealed class TapeSide
    {
        public TapeSide(
            bool isRecorded,
            string name,
            DateTime recordedAt,
            string deckName,
            string inputName,
            int peakLevel,
            string noiseReduction,
            bool hx,
            bool mpx,
            bool dubbed,
            string speed,
            int bias,
            int biasCalibration,
            string equalization,
            decimal level,
            decimal levelCalibration,
            string contents,
            string artist,
            string title)
        {
            IsRecorded = isRecorded;
            Name = name;
            RecordedAt = recordedAt;
            DeckName = deckName;
            InputName = inputName;
            PeakLevel = peakLevel;
            NoiseReduction = noiseReduction;
            Hx = hx;
            Mpx = mpx;
            Dubbed = dubbed;
            Speed = speed;
            Bias = bias;
            BiasCalibration = biasCalibration;
            Equalization = equalization;
            Level = level;
            LevelCalibration = levelCalibration;
            Contents = contents;
            Artist = artist;
            Title = title;
        }

        public bool IsRecorded { get; }
        public string Name { get; }
        public DateTime RecordedAt { get; }
        public string DeckName { get; }
        public string InputName { get; }
        public int PeakLevel { get; }
        public string NoiseReduction { get; }
        public bool Hx { get; }
        public bool Mpx { get; }
        public bool Dubbed { get; }
        public string Speed { get; }
        public int Bias { get; }
        public int BiasCalibration { get; }
        public string Equalization { get; }
        public decimal Level { get; }
        public decimal LevelCalibration { get; }
        public string Contents { get; }
        public string Artist { get; }
        public string Title { get; }

        public static TapeSide Empty()
        {
            return new TapeSide(
                false,
                string.Empty,
                DateTime.MinValue,
                string.Empty,
                string.Empty,
                0,
                string.Empty,
                false,
                false,
                false,
                string.Empty,
                0,
                0,
                string.Empty,
                decimal.Zero,
                decimal.Zero,
                string.Empty,
                string.Empty,
                string.Empty);
        }
    }
}
